### ISSUE TITLE
Principled Shader

### ISSUE THREAD

**Issue by Theverat** [LuxCoreRender/LuxCore#91]
This shader is apparently artist-friendly, physically based and compatible with many other applications (substance painter creates texture maps for it, game engines also support it etc.).
Since Cycles supports it, it would also make a material converter a bit easier to write.

**Comment by canonex** [comment]
More than this, artist used to deal with this shader can switch render engine with a little pain!
Thank you,
Riccardo

**Comment by JayFoxRox** [comment]
https://github.com/LuxCoreRender/LuxCore/issues/175#issuecomment-474775461 might be relevant to this issue.

@Dade916 Are principled shaders planned for v2.3? What's a rough estimated date for v2.3 (months / years)?

Principled shaders have become the defacto-standard for materials, so I'd assume a rework would include them.

This could affect my application, as I'm about to recreate some materials from Principled Shader (used in Cycles and Game-Engine) using LuxCore materials; so if LuxCore will support these original shaders soon, I will shift my focus to other code-aspects of my application for now.

**Comment by Dade916** [comment]
> @Dade916 Are principled shaders planned for v2.3?

Yes, it is planned for v2.3 (https://forums.luxcorerender.org/viewtopic.php?f=5&t=308)

> What's a rough estimated date for v2.3 (months / years)?

I assume v2.2 will be finished before the summer and v2.3 before the end of the year. An alpha/beta with a Walt Disney Principled Shader material is likely to be available in Autumn.

There is some preliminary work for v2.3 to do before implementing new materials: implementing the Principled Shader material alone is not a big issue. It is just more logical to do it after all preliminary work.

**Comment by lacilaci** [comment]
I think this issue should be updated, we have disney but it needs upgrade to support transmission. Ideally with it's own translucent/thin mesh mode, microroughness control to control reflection glossiness on grazing angles and irridescent film controls. So that we can use this as ubermaterial for most common materials.

**Comment by Dade916** [comment]
SIGGRAPH 2015 Disney material extensions still to add.

**Comment by Dade916** [comment]
Some relevant information about Disney SIGGRAPH 2015 extensions.

Extending the Disney BRDF to a BSDF with Integrated Subsurface Scattering (Brent Burley):

https://blog.selfshadow.com/publications/s2015-shading-course/burley/s2015_pbs_disney_bsdf_slides.key
https://blog.selfshadow.com/publications/s2015-shading-course/burley/s2015_pbs_disney_bsdf_slides.pdf
https://blog.selfshadow.com/publications/s2015-shading-course/burley/s2015_pbs_disney_bsdf_notes.pdf
